### PR TITLE
Fix depth check in 3D mode

### PR DESCRIPTION
**Pull request by ziyaointl**
This PR consists two parts:
1. fix depth check
2. fix z-fighting

For more details, see #152

### CODE REVIEW

**Comment by parsoyaarihant** [comment]
How about rendering the lines after the triangles for a given shape?

Then we wont have to do the EPSILON transformation right?

**Comment by parsoyaarihant** [comment]
@ziyaointl , I have added a comment in renderer3d.py 

Previously I have observed that Github does not give notification for comments on file.

File: p5/sketch/renderer3d.py [hunk 1]
```diff
@@ -293,11 +298,19 @@ def add_to_draw_queue(self, stype, vertices, edges, faces, fill=None, stroke=Non
 	def flush_geometry(self):
 		"""Flush all the shape geometry from the draw queue to the GPU.
 		"""
-		current_shape = None
 		current_queue = []
+		# line_transform is used whenever we render lines to break ties in depth
+		# We transform the points to camera space, move them by Z_EPSILON, and them move them back to world space
+		line_transform = inv(self.lookat_matrix).dot(translation_matrix(0, 0, Z_EPSILON).dot(self.lookat_matrix))
```
[the remark this answers is not in the document]
Comment (ziyaointl) [reply p5/sketch/renderer3d.py]: Good point! I moved the transform inside the conditional.

**Comment by parsoyaarihant** [comment]
Tip: In your commit, if you mention "Fixes #Issue_number". That issue will automatically get closed after the branch is merged.

**Pull request merged**